cordova-lib/src/cordova/plugin.js

```diff
@@ -157,6 +157,7 @@ module.exports = function plugin(command, targets, opts) {
 
                             if (missingVariables.length) {
                                 shell.rm('-rf', pluginInfo.dir);
+                                events.emit('verbose', 'Removing ' + pluginInfo.dir + ' due to installation failure');
                                 var msg = 'Variable(s) missing (use: --variable ' + missingVariables.join('=value --variable ') + '=value).';
```

Comment:
Not sure if i understand you correctly. The `events.emit` at L160 is added because the reporter was confused by the abscence of plugin inside of `<project>/plugins` directory so i think it'd be better if we clarify that we've removed that.

The missing variable is a part of error message and is logged by caller code as a part of rejection handling (on particular in [restore-utilL127](https://github.com/apache/cordova-lib/pull/420/files#diff-e9221ac2a69074d78c7fb9a8e6b9d4d6R127))